src/net_processing.cpp

```diff
@@ -798,6 +803,11 @@ void PeerLogicValidation::BlockChecked(const CBlock& block, const CValidationSta
                 Misbehaving(it->second.first, nDoS);
         }
     }
+    else if (state.IsValid() && !IsInitialBlockDownload() && mapBlocksInFlight.empty()) {
```

Comment:
I don't think this check of mapBlocksInFlight.empty() will play nicely with #9352 which doesn't MarkBlockReceived until after ProcessNewBlock